src/libs/actions/App.js

```diff
@@ -206,6 +215,39 @@ function setUpPoliciesAndNavigate(session) {
         });
 }
 
+function openProfile() {
+    const oldTimezoneData = myPersonalDetails.timezone || {};
+    const newTimezoneData = {
+        automatic: oldTimezoneData.automatic || true,
```

Comment:
:sus: if `oldTimezoneData.automatic` is `false`, this will be `true`, it it was `true`, this will be `true`, right? Is this how we want it? we should probably refactor

Reply:
Aah trueee I was thinking `??` but I don't think we like using that in our app code 😅 good catch :D

I think I can use `lodashGet` here like this:

<img width="326" alt="Screen Shot 2022-07-28 at 11 50 14 AM" src="https://user-images.githubusercontent.com/3885503/181476720-06e51049-d2bd-4617-af10-dd025c3a6bed.png">

Reply:
What is the automatic actually? Isn't this automatic all the time when we call open profile and get the timezone from `moment.guess`? Should it not be marked as `automatic: true` in this case everytime?

So many questions haha

Reply:
Soooooooo `automatic` is the setting if the user wants to have the timezone update automatically or not.

The related Web-E logic is here: https://github.com/Expensify/Web-Expensify/blob/ae9d6761900b4518c1f10106ef61e27138362b97/lib/UserAPI.php#L3227-L3233

Basically, we get the current timezone NVP value. If the timezone's `automatic` setting is `true`, we update the NVP to whatever the new timezone value is. If it's `false`, we don't do anything